### DISCUSSION TITLE
How to use environment variable into self-hosted runner label?

### DISCUSSION THREAD

**Discussion by vitalyk-multinarity** [community/community#31221] [Actions] [answered]
I'm launching self-hosted runner for different repos, using the following name convention: 

` github-actions-runner-_repo_name_-${GITHUB_RUN_NUMBER} `

(I cannot use github.repository 'as is' into hostname because GCP doesn't allow capital letters)

And when I'm using them as 

`runs-on: [ self-hosted, "github-actions-runner-my_repo_name-${{ github.run_number }}" ]`

As far as 'my_repo_name' is hardcoded, it works, but if I'm trying to use 

`runs-on: [ self-hosted, "github-actions-runner-${{ env.RUN_NAME }}-${{ github.run_number }}" ]`

I'm getting  error

`The workflow is not valid. .github/workflows/self-hosted-tests.yml (Line: 38, Col: 29): Unrecognized named-value: 'env'.`

Am I missing something? Or we cannot use environment variables into labels?

**Comment by airtower-luna** [comment, marked as answer]
> Or we cannot use environment variables into labels?

Exactly. The [`env` context documentation](https://docs.github.com/en/actions/learn-github-actions/contexts#env-context) says:

> You can use the env context in the value of any key in a step except for the id and uses keys.

`runs-on` isn't part of any step.

**Reply by vitalyk-multinarity** [reply]
Thank you!
So there is no way to avoid hardcoded labels...
